hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/contracts/operation/HederaCreateOperationExternalizer.java

```diff
@@ -153,7 +154,7 @@ private void finalizeHollowAccountIntoContract(AccountID hollowAccountID, Hedera
         // update the hollow account key to be the default contract key
         updater.trackingAccounts().set(hollowAccountID, KEY, STANDIN_CONTRACT_ID_KEY);
 
-        // set initial contract nonce to 1
-        updater.trackingAccounts().set(hollowAccountID, ETHEREUM_NONCE, 1L);
+        // set initial contract nonce to be the same as the nonce of the created account
+        updater.trackingAccounts().set(hollowAccountID, ETHEREUM_NONCE, contractNonce);
```

Comment:
Q: could this ever not be 1?
A hollow account would have a 0 nonce. The only way for it to become 1 would be to submit an `EthereumTransaction` in which case it would become a full account and could then not become a contract.
In the above logic the `updater.get(childFrame.getContractAddress()).getNonce()` is for a newly created account so it should always return 1 right?
So wasn't the previous logic better where it's set to 1 as any new contract would be and there's no extra query required?